drakrun/drakrun/draksetup.py

```diff
@@ -551,19 +554,20 @@ def postinstall(report, generate_usermode):
         f.write(runtime_info.to_json(indent=4))
 
     logging.info("Saving VM snapshot...")
-    subprocess.check_output('xl save vm-0 ' + os.path.join(VOLUME_DIR, "snapshot.sav"), shell=True)
+    subprocess.check_output('xl save -c vm-0 ' + os.path.join(VOLUME_DIR, "snapshot.sav"), shell=True)
 
-    storage_backend = get_storage_backend(install_info)
     storage_backend.snapshot_vm0_volume()
     logging.info("Snapshot was saved succesfully.")
 
     if generate_usermode:
         try:
-            create_rekall_profiles(install_info)
+            create_rekall_profiles(install_info, runtime_info, kernel_profile)
         except RuntimeError as e:
             logging.warning("Generating usermode profiles failed")
             logging.exception(e)
 
+    subprocess.check_output('xl destroy vm-0', shell=True)
+
```

Comment:
I think it woud be a good idea to create a VM object in this function to handle saving and destroying the VM.

Reply:
yeah, I was thinking about it while working on VM unit tests. I have more refactoring ideas also which I wanted to discuss regarding the `vm.py`. That PR will address `vm.py` related stuff. It will also be easy to track these refactoring changes that way ig. Will do this in that PR if it seems fine?

Reply:
:+1: